gis/text_to_centreline/sql/function-get_lines_btwn_interxn.sql

```diff
@@ -25,23 +27,47 @@ AS $BODY$
 BEGIN
 RETURN QUERY
 
-WITH 
-results AS (SELECT _int_start, _int_end, * FROM
-    pgr_dijkstra(format('SELECT id, source::int, target::int,
-                 CASE WHEN levenshtein(TRIM(lf_name), TRIM(%L), 1, 1,1) < 3 THEN (0.3*cost)::float ELSE cost END AS cost 
-                 from gis.centreline_routing_undirected_lfname'::text, _highway2),
-                 _int_start::bigint, _int_end::bigint, FALSE)
+WITH results AS (
+    SELECT
+        _get_lines_btwn_interxn._int_start,
+        _get_lines_btwn_interxn._int_end,
+        dijkstra.seq,
+        dijkstra.edge
+    FROM pgr_dijkstra(format('SELECT
+                centreline_id AS id,
+                from_intersection_id AS source,
+                to_intersection_id AS target,
+                CASE WHEN levenshtein(TRIM(linear_name_full), TRIM(%L), 1, 1,1) < 3 THEN (0.3*costs.cost)::float ELSE costs.cost END AS cost 
+            FROM gis_core.centreline_latest,
+            LATERAL (
```

Comment:
do we need lateral here? Can it be directly used in the case when ?

Reply:
Rewrote without lateral!